### ISSUE TITLE
ValidationError.data has wrong type

### ISSUE THREAD

**Issue by swenzel-arc** [payloadcms/payload#2764]
### Link to reproduction

none

### To Reproduce

```ts
import { ValidationError } from "payload/errors";

const err = new ValidationError([{ message: "test", field: "test" }]);

type ValidationErrorData = { message: string, field: string }[]
const validationErrors: ValidationErrorData = err.data;  // Typescript error
```

### Describe the Bug

`err.data` is not assignable to `validationErrors` because the former is defined as `{ [key: string]: unknown;}` in [ExtendableError](https://github.com/payloadcms/payload/blob/master/src/errors/APIError.ts#L7) which ValidationError ultimately inherits from.

However [ValidationError](https://github.com/payloadcms/payload/blob/master/src/errors/ValidationError.ts#LL11) is passing `results` (which has type `{ message: string, field: string }[]`) as value for `data` to its parent's constructor.

### Payload Version

1.8.3

**Comment by denolfe** [comment]
Thanks for the report @swenzel-arc . While this is not ideal, the initial intent of having data typed as `{ [key: string]: unknown;}` was to make it as flexible as possible. My concern with changing this, is that it would likely be a breaking change.

Open to suggestions on how we could accommodate this but also preventing a breaking change 👍

**Comment by denolfe** [comment]
@swenzel-arc Good suggestion. I've implemented it in #2864.
